### ISSUE TITLE
Glue: Create transformations editor

### ISSUE THREAD

**Issue by ifrost** [grafana/grafana#62095]
### Background

Transformations functionality was added in https://github.com/grafana/grafana/issues/60023. 
### Requirements ###

High level: _Allow adding / editing / removing transformations in UI_

#### Technical details

Transformations editor is a simple list allowing adding multiple transformations to a single correlation.

<img width="644" alt="Screenshot 2023-03-03 at 16 18 06" src="https://user-images.githubusercontent.com/745532/222757652-1cf04b04-d3e0-461c-9c04-37c2a3af6809.png">

(This is just a mock, feel free to use different columns/UI)

<details><summary>To be confirmed: Expected structure</summary>

```json
{
  "type": "query",
  "field": "[string]",
  "target": "[object]",
  "transformations": [
    {
      "type": "regex",
      "mapValue": "[string]",
      "expression": "[string]"
    },
    {
      "type": "logfmt",
      "variable": "[string]"
    }
  ],
}
```

</details>

Clicking on "New transformation" opens up a model with following options:

* Dropdown to select type of the transformation (`regex`, `logfmt`) [required]
* Field name input [optional] (to be changed into a dropdown when a sample is selected in https://github.com/grafana/grafana/issues/60965). If not selected, the target field for the correlation is used (this logic is already in place, but we should be explicit about it, maybe in a tooltip)
* Transformation options [required]:
  * `regex`: 
    * input for regex expression [required]
    * name of the variable [optional] (if empty name of the field is used). There should be a tooltip explaining that capture groups can be used
  * `logfmt`: no options
* Validation: only required fields are validated. We can add more validation rules once sample value is provided.

Each row can be removed. No extra confirmation is needed.

Changes are applied only once entire correlation is saved.

Each row can be edited by clicking on a row. It opens the same form as for adding.

#### To Do
- [ ] Implement adding/editing/removing transformations
- [ ] Validation
- [ ] Removing doesn't need extra confirmation
- [ ] The list can be represented as a table, each row representing a transformation (type, field name, options, variable name in separate columns, each is a single input)
- [ ] We need to update documentation on how to use transformations. (Depends on https://github.com/grafana/grafana/issues/62100). It should cover regexp and logfmt with examples.
- [ ] Track how many transformations are added
- [ ] Create a dashboard to track how many and of what type transformations are being added (compare with number of correlations added)

Questions:
* Should removing require extra confirmation? No, because the correlation has to saved anyway to apply the changes.

<details><summary>DoR checklist</summary>

- [ ] Should our squad work on it?  ***Yes, we're the owner of correlations***
- [ ] Why is it important?  ***Without it transformations will be available only with provisioning which is not accessible to all users***
- [ ] Dependencies: ***No***
- [ ] Priority: ***Medium***
- [ ] Functional requirements. ***Defined.***
- [ ] Non-functional requirements . ***N/A***
- [ ] Documentation requirements. ***Defined***
- [ ] UX design / approval. ***Feedback on the PR***
- [ ] Technical design / approval.  ***Questions pending***
- [ ] How will the outcome value of the feature be measured? ***transformations created / correlations created ratio***
- [ ] Should we write a blog post about it?  ***No. It will be done as a separate task when correlations are released.***
</details>
